src/main/java/uk/gov/hmcts/sptribs/common/event/page/SubjectDetails.java

```diff
@@ -21,4 +28,25 @@ public void addTo(PageBuilder pageBuilder) {
             .mandatory(CicCase::getEmail, "cicCaseContactPreferenceType = \"Email\"")
             .done();
     }
+
+    private AboutToStartOrSubmitResponse<CaseData, State> midEvent(
+        CaseDetails<CaseData, State> details,
+        CaseDetails<CaseData, State> detailsBefore) {
+        final CaseData data = details.getData();
+        final List<String> errors = new ArrayList<>();
+
+        if (null != data.getCicCase() && null != data.getCicCase().getAddress()) {
+            if (StringUtils.isEmpty(data.getCicCase().getAddress().getCountry())) {
+                errors.add("Country is mandatory");
```

Comment:
If Address is null, does that mean there should be an error added?

Reply:
No, it can address or email. We need to check for Country/post code only if Address is selected.